tmt/steps/report/junit.py

```diff
@@ -58,40 +67,230 @@ def duration_to_seconds(duration: Optional[str]) -> Optional[int]:
+        """ Possibility to iterate over results by iterating an instance """
+        return iter(self._results)
+
+    def __len__(self) -> int:
+        """ Returns the number of results """
+        return len(self._results)
+
+    @functools.cached_property
+    def passed(self) -> list[ResultWrapper]:
+        """ Returns results of passed tests """
+        return [r for r in self._results if r.result == ResultOutcome.PASS]
+
+    @functools.cached_property
+    def skipped(self) -> list[ResultWrapper]:
+        """ Returns results of skipped tests """
+        return [r for r in self._results if r.result == ResultOutcome.INFO]
+
+    @functools.cached_property
+    def failed(self) -> list[ResultWrapper]:
+        """ Returns results of failed tests """
+        return [r for r in self._results if r.result == ResultOutcome.FAIL]
+
+    @functools.cached_property
+    def errored(self) -> list[ResultWrapper]:
+        """ Returns results of tests with error/warn outcome """
+        return [r for r in self._results if r.result in [
+                ResultOutcome.ERROR,
+                ResultOutcome.WARN]]
+
+    @functools.cached_property
+    def duration(self) -> int:
+        """ Returns the total duration of all tests in seconds """
+        durations: list[int] = [duration_to_seconds(r.duration) or 0 for r in self._results]
+        return sum(durations)
+
+
 def make_junit_xml(
-        report: 'ReportPlugin[ReportJUnitData]|ReportPlugin[ReportPolarionData]'
-        ) -> 'junit_xml.TestSuite':
-    """ Create junit xml object """
-    junit_xml = import_junit_xml()
+        phase: tmt.steps.report.ReportPlugin[Any],
+        flavor: str = DEFAULT_FLAVOR_NAME,
+        template_path: Optional[Path] = None,
+        include_output_log: bool = True,
+        prettify: bool = True,
+        results_context: Optional[ResultsContext] = None,
+        **extra_variables: Any
+        ) -> str:
+    """
+    Create JUnit XML file and return the data
+
+    :param phase: instance of a ReportPlugin.
+    :param flavor: name of a JUnit flavor to generate.
+    :param template_path: if set, the provided template will be used instead of
+        a pre-defined flavor template. In this case, the ``flavor`` must be set
+        to ``custom`` value.
+    :param include_output_log: if enabled, the ``<system-out>`` tags are included
+        in the final template output.
+    :param prettify: allows to control the XML pretty print.
+    :param results_context: if set, the provided ResultsContext is used in a template.
+    :param extra_variables: if set, these variables get propagated into the
+        Jinja template.
+    """
+
+    # Get the template context for TMT results
+    results_context = results_context or ResultsContext(phase.step.plan.execute.results())
+
+    # Prepare the template environment
+    environment = default_template_environment()
+
+    template_dir = Path('steps/report/junit/templates/')
```

Comment:
I'd promote this to module-level "constant", something like `DEFAULT_TEMPLATE_PATH` or something similar. It's an important string constant, hidden in the code.